### ISSUE TITLE
Segment index incorrectly parsed as hex

### ISSUE THREAD

**Issue by andersmelander** [andersmelander/map2pdb#4]
**[Original report](https://bitbucket.org/anders_melander/map2pdb/issues/4) by Anders@Melander (Bitbucket: [Anders](https://bitbucket.org/Anders), GitHub: [Anders](https://github.com/Anders)).**

----------------------------------------

From: [https://en.delphipraxis.net/topic/4853-map2pdb-profiling-with-vtune/?do=findComment&comment=68854](https://en.delphipraxis.net/topic/4853-map2pdb-profiling-with-vtune/?do=findComment&comment=68854)

---

Excellent work ... Thanks !

I was using the tool with some map file and I got an AV error. I started investigating and I found a bug in the map parser﻿ : 

```pas
// debug.info.reader.map.pas (line 349):
var SegmentID: Cardinal := HexToInt16(Reader.LineBuffer, n);
```

The SegmentID should be read as a DECIMAL value and not as a HEX ! if the map file contains more than 9 segment, the next segment is emitted like \(0010:xxxxx\) but you're reading it like a hex \(so its ID becomes 16 !\) and then there is this line 

```pas
FSegments[ListIndex] := Result; // FSegments[10 .. ListIndex - 1] = nil => AV
```

 A simple fix : 

```pas
function DecToInt32(const s: string; var Offset: integer): integer;
var
  P: PChar;
begin
  Result := 0;
  P := PChar(s);
  while Ord(P^) in [$30 .. $39] do
  begin
    Result := (Result * 10) + (Ord(P^) - $30);
    Inc(P);
    Inc(Offset);
  end;
end;
// ...
var SegmentID: Cardinal := DecToInt32(reader.LineBuffer, n);
```

‌

**Comment by andersmelander** [comment]
**Original comment by Anders Melander (Bitbucket: [anders_melander](https://bitbucket.org/anders_melander), ).**

----------------------------------------

Segment index values are now parsed as decimal integers and logged as same.
Fixes #4

**Comment by andersmelander** [comment]
**Original comment by Anders Melander (Bitbucket: [anders_melander](https://bitbucket.org/anders_melander), ).**

----------------------------------------

Segment index values are now parsed as decimal integers and logged as same.
Fixes #4
